test/e2e_node/conformance_test.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_node
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/errors"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+type ConformanceCase struct {
+	Container api.Container
+	//test both the valid/invalid images
+	Status string
+}
+
+var _ = Describe("Kubelet", func() {
+	var cl *client.Client
+	var testcases []ConformanceCase
+
+	BeforeEach(func() {
+		// Setup the apiserver client
+		cl = client.NewOrDie(&client.Config{Host: *apiServerAddress})
+		testcases = []ConformanceCase{
+			{
+				Container: api.Container{
+					Image:           "gcc.io/google_containers/busybox",
+					Name:            "busybox",
+					Command:         []string{"echo", "'Hello World'"},
+					ImagePullPolicy: api.PullIfNotPresent,
+				},
+				//terminated soon.
+				Status: "terminated",
+			},
+			{
+				Container: api.Container{
```

Comment:
For test clarity, I would pull these into separate contexts that explain what they are doing instead of trying to generalize with a struct:

Context("when running a pod that terminates") {
  It("It should start successfully"){...}
  It("It should report its status as 'terminated'"){}
  It("It should be possible to delete")
}

Context("when running a pod with an container invalid image") {
  It("It should start successfully"){...}
  It("It should report its status as 'waiting'"){}
  It("It should be possible to delete")
}